main.js

```diff
@@ -62,17 +59,55 @@ function render(file) {
         camera.position.setZ(30);
     }
 
-    function loadShape() {
-        let shape = createFaceGeom();
-
-        scene.add(shape);
+    function loadShapes() {
+        // clear the scene of existing objects
+        scene.clear();
+        let shapes = [];
 
-        const edges = new THREE.EdgesGeometry(shape.geometry);
-        const lines = new THREE.LineSegments(edges, new THREE.LineBasicMaterial({color: 0xffffff}));
-        scene.add(lines);
+        let vertex_coords = fold["vertices_coords"];
+        if (vertex_coords) {
+            shapes.push(createFaceGeom(vertex_coords));
```

Comment:
I think this will break now that `createFaceGeom` expects a second argument. I think the easiest way to do this if is to look for the two different formats and then work with either the frame if we find the appropriate one, or the top-level object if there is no frame.